### ISSUE TITLE
Ready is triggered before the library is loaded

### ISSUE THREAD

**Issue by nachitox** [muicss/loadjs#73]
daterangepicker.js is loaded after moment, which loads after jquery.

![load](https://user-images.githubusercontent.com/1167303/43598693-b0292e58-965b-11e8-93b4-d9922f15967d.png)

Sometimes it works fine, but other times daterangepicker does not exists.

![error](https://user-images.githubusercontent.com/1167303/43598702-b436aa7a-965b-11e8-9709-9a7508c03046.png)

I'd prefer waiting a few ms than the library throwing an error.

What can I do?
Thank you

**Comment by amorey** [comment]
It looks like the screenshot got cut off so I can't see how jQuery is being loaded. Can you share the code that is causing the error?

**Comment by nachitox** [comment]
jquery and moment are loaded using loadjs too

```
loadjs('https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js', 'jquery');
loadjs.ready('jquery', function() {
loadjs([REAL_PATH + 'moment/moment.min.js',], 'moment');
});
```

**Comment by amorey** [comment]
I don't know the internals of daterangepicker.js but it's possible that it has a dependency on `DOMContentLoaded` that is causing the intermittent failure. Have you tried wrapping your code with `jQuery.ready()`?
```javascript
loadjs.ready('datepicker', function() {
  $.ready(function() {
    $('#datefrom').daterangepicker();
  });
});
```

In addition, your code appears to be loading jQuery, moment.js and daterangepicker.js sequentially. If you want to load them in parallel and execute them in order you can use `async: false`:
```javascript
loadjs(['/path/to/jquery.js', '/path/to/moment.js', '/path/to/daterangepicker.js'], {
  async: false,
  success: function() {},
  error: function (pathsNotFound) {}
});
```

**Comment by saurabhnanda** [comment]
I'm seeing similar behaviour.

**Comment by amorey** [comment]
Did you try the suggestions above? https://github.com/muicss/loadjs/issues/73#issuecomment-410404667

**Comment by Rainy-Bear** [comment]
I'm seeing similar behaviour also. Miraculously, there is no problem in the my local PC, but there are occasional errors on the line

**Comment by Rainy-Bear** [comment]
addtionally, I have seted the async is ture

**Comment by amorey** [comment]
The javascript object accepts `false` as a value so if you set the async property to false then the scripts will not be loaded asynchronously (https://www.w3schools.com/jsref/prop_script_async.asp).

Can you share the problematic code so I can take a closer look at it?

**Comment by Rainy-Bear** [comment]
like it
`if(hasLoad) {
    loadjs(['/path/to/map.js'], 'map' {
      async: false
    });
    hasLoad = true;
}
loadjs.ready(['map'],()=>{
 map.refresh(); // sometime map is undefined
})
`
In particular, it will be called twice during initialization, first is 'componentDidMount' second 'componentWillReceiveProps'

**Comment by amorey** [comment]
Without knowing what map.js does internally I can't tell what the problem is. There could be a dependency on `DOMContentLoaded` that is causing an intermittent failure as described above. Can you provide a working code snippet that I can run? If you can provide a working HTML file that demonstrates the problem that would make it easier to debug.

**Comment by DoubleCorner** [comment]
@amorey the map.js is here https://github.com/huiyan-fe/mapv/blob/master/build/mapv.min.js

**Comment by amorey** [comment]
@DoubleCorner I got an email notification that you left a new message but now I can't see the message. Did you figure out the problem?

**Comment by DoubleCorner** [comment]
@amorey no, I will try again. if I figure out, I will write my solution.
